### PR TITLE
Fix analysis script to run from anywhere

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

Fixes a small problem with the analyze.dart script that fixes it so that it can be invoked from any directory, not just the Flutter root.

## Tests

- Added a test for the localizations test, since there wasn't one (which is why this bug appeared).

### CODE REVIEW

**Review by craiglabenz — APPROVED** [review]
Verified by running the analysis script from within the repo, so this LGTM

**Comment by craiglabenz** [comment]
Oh goodie, a Windows-only error.

File: dev/bots/test/analyze-test-input/root/dev/tools/localization/bin/gen_localizations.dart [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  print('''
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+void main(List<String> args) {
+  print('Expected output $type');
+}
+''');
+}
```
Comment (dnfield) [inline dev/bots/test/analyze-test-input/root/dev/tools/localization/bin/gen_localizations.dart:22]: nit: newlines at EOF here and below

Comment (gspencergoog) [reply dev/bots/test/analyze-test-input/root/dev/tools/localization/bin/gen_localizations.dart]: Fixed, thanks.

File: dev/bots/test/analyze_test.dart [hunk 2]
```diff
@@ -38,7 +38,8 @@ Future<String> capture(AsyncVoidCallback callback, { int exitCode = 0 }) async {
 
 void main() {
   final String testRootPath = path.join('test', 'analyze-test-input', 'root');
-  final String dartPath = path.canonicalize(path.join('..', '..', 'bin', 'cache', 'dart-sdk', 'bin', Platform.isWindows ? 'dart.exe' : 'dart'));
+  final String dartName = Platform.isWindows ? 'dart.exe' : 'dart';
```
Comment (craiglabenz) [inline dev/bots/test/analyze_test.dart:41]: the `.exe` extension on Windows always trips me up 😆

**Pull request merged**